### DISCUSSION TITLE
Can't develop with poetry 1.2.0+ on remote servers (i.e., using ssh)

### DISCUSSION THREAD

**Discussion by choppsv1** [python-poetry/poetry#7079] [Q&A] [answered]
I'm am having trouble with poetry >= 1.2.0 in what I would think was a very common case. It's so common that I'm thinking I'm doing something wrong. I do not have this problem with 1.1.15.

- I ssh to my server.
- I pip install poetry.
- I run poetry and it fails.

 By fails I mean e.g., `poetry add pexpect` in my brand new project as first dependency add) fails b/c of some dbus error:

```
  DBusErrorResponse
  [org.freedesktop.DBus.Error.UnknownMethod] ('Object does not exist at path “/org/freedesktop/secrets/collection/login”',)
...
```

Underneath AFAICT what is going on is that poetry is using keyring (for who knows why, given this isn't a publish event...) keyring is launching gnome-keyring, and then failing b/c I'm logged in over ssh and there is *no* dbus running.

If I manually uninstall keyring `pip uninstall keyring` poetry works just fine.

I can hardly believe I'm the only person who does their work on remote servers over ssh, so any idea what else I could be doing wrong here? And why is poetry using the keyring in the first place (for a non-publish action)?

Thanks,
Chris.

**Comment by dimbleby** [comment]
if you search the issue tracker you will find that you are not the only person, and some possible workarounds

**Comment by neersighted** [comment, marked as answer]
Keyring is used for many operations, like fetching metadata and packages. We need to always check for credentials as they may be required for accounting/resource limitations at custom indexes, or a PyPI proxy. The current experience is very poor as real world testing has revealed many configurations the current error handling does not account for, but that were not revealed by CI/developer machines, likely because we tend to treat our machines as disposable and blow them away/keep them paired down.

In any case, see https://github.com/python-poetry/poetry/issues/1917 for the current workarounds, as well as some partially reviewed PRs working to address this.

**Comment by choppsv1** [comment]
Thanks for the pointer. Wish I had time to help with a fix.
